### PR TITLE
arch/tricore: add Infineon AURIX TriCore support

### PR DESCRIPTION
**Pull request by anchao**
## Summary

arch/tricore: add Infineon AURIX TriCore support
Add support for tricore TC397

1. Porting based on AURIX TC397 KIT_A2G_TC397_5V_TFT evaluation board

   https://www.infineon.com/cms/en/product/evaluation-boards/kit_a2g_tc397_5v_tft/

2. In order to avoid license and coding style issues, The chip-level code still uses the implementation of AURIX Development Studio SDK. The SDK package will be downloaded as a third-party package during compilation:

   https://github.com/anchao/tc397_sdk

3. Single core only, SMP implementation will be provided in the future.
4. Implemented the basic System Timer, ASCLIN UART driver.
5. Only the Tasking tool chain is supported (ctc/ltc, license maybe required)
6. 'ostest' can be completed on the TC397 development board.

How to run?

1. Setup the tasking toolchain and license

```
$ export TSK_LICENSE_KEY_SW160800=d22f-7473-ff5d-1b70
$ export TSK_LICENSE_SERVER=192.168.36.12:9090
```

2. Build nuttx ELF

```
$ ./tools/configure.sh tc397/nsh
$ make -j
...
artc I800: creating archive libc_fpu.a
LD: nuttx
```

3. Switch to windows PC, setup AURIX-studio to Debugger Launcher

4. Replace running ELF to nuttx, and re-download ELF

5.  Config terminal on IDE

6. Run nuttx

## Impact

N/A

## Testing

Tricore KIT_A2G_TC397_5V_TFT

### CODE REVIEW

**Comment by acassis** [comment]
@anchao Did you consider verifying with Infineon if they can allow to use their code under Apache license. Many companies are starting to use to shortcut to just download the vendor SDK/HAL, but I think this approach could be risk. Also a proper port based on datasheet will avoid license issues. Although it is slower, this is how Greg did all the ports!

**Comment by anchao** [comment]
@acassis 

1. All files in this pull request are **re-implemented based** on nuttx RTOS and do not use any code from Infineon SDK/HAL.
2. The Infineon **low-level driver (iLLD)** referenced in the nuttx kernel is released for free in AURIX™ Development Studio. Anyone can obtain the relevant SDK on the infineon official website:

https://www.infineon.com/cms/en/product/promopages/aurix-development-studio/




and Infineon also open sourced the relevant code on github:

https://github.com/Infineon/AURIX_code_examples/tree/master

e.g:
https://github.com/Infineon/AURIX_code_examples/tree/master/code_examples/GPT12_Timer_Concatenation_1_KIT_TC397_TFT:

4. nuttx only provides a free platform licensed by Apache to demo the Infineon development board. If developers or companies want to use it commercially, then they should consider licensing issues.

**Comment by anchao** [comment]
@acassis I have added  TriCore architecture to documentation, which mentioned the reference the core chip implement reused Infineon low-level driver (iLLD) SDK/HAL, please review

File: arch/tricore/Kconfig [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	bool
+	default n
+
+config ARCH_FAMILY
+	string
+	default "tc3xx"		if ARCH_TC3XX
+
+config ARCH_CHIP
+	string
+	default "tc3xx"		if ARCH_CHIP_TC397
```
Comment (xiaoxiang781216) [inline arch/tricore/Kconfig:29]: tc397

File: arch/tricore/include/inttypes.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define _PRI32PREFIX "l"
```
Comment (xiaoxiang781216) [inline arch/tricore/include/inttypes.h:32]: why not directly use "l" or "ll"

File: arch/tricore/src/.gitignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+/locked.r
```
Comment (xiaoxiang781216) [inline arch/tricore/src/.gitignore:1]: why need this entry

File: arch/tricore/src/common/Toolchain.defs [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+#   3                               alias for --fp-model=cflnrSTz (fast-sp)
+
+ARCHOPTIMIZATION += --fp-model=2
+LDFLAGS          += --fp-model=2
+LDFLAGS          += -lfp_fpu
+
+LDFLAGS          += --hex-format=s -Wl-OtxYcL -Wl-mcrfiklsmnoduq
+LDFLAGS          += -lrt
+
+# ctc W500: ["stdio/lib_libvsprintf.c" 884/29] expression without effect
```
Comment (xiaoxiang781216) [inline arch/tricore/src/common/Toolchain.defs:107]: should we fix these warning instead disabling them

File: arch/tricore/src/tc3xx/Kconfig [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+comment "TC3XX Configuration Options"
+
+menu "TC3XX Peripheral Support"
+
+# These "hidden" settings determine whether a peripheral option is available
+# for the selected MCU
+
+config TC3XX_HAVE_UART0
+	bool
+	default y
```
Comment (xiaoxiang781216) [inline arch/tricore/src/tc3xx/Kconfig:15]: remove, should select by other option

**Review by acassis — APPROVED** [review]

File: arch/tricore/src/.gitignore (hunk 3 above)
Comment (anchao) [reply arch/tricore/src/.gitignore:1]: let me remove

File: arch/tricore/src/tc3xx/Kconfig (hunk 5 above)
Comment (anchao) [reply arch/tricore/src/tc3xx/Kconfig:15]: Done

File: arch/tricore/src/common/Toolchain.defs (hunk 4 above)
Comment (anchao) [reply arch/tricore/src/common/Toolchain.defs:107]: Most of the warnings are compiler defects, which have been fixed in GNU toolchain

File: arch/tricore/include/inttypes.h (hunk 2 above)
Comment (anchao) [reply arch/tricore/include/inttypes.h:32]: Done

**Comment by xiaoxiang781216** [comment]
Let's ignore the ci temp break which is fixed by https://github.com/apache/nuttx/pull/11740.

**Pull request merged**